polaris/tasks/ocean/seamount/forward.py

```diff
@@ -0,0 +1,158 @@
+    """
+    A step for performing forward MPAS-Ocean runs as part of seamount
+    test cases.
+
+    Attributes
+    ----------
+    task_name : str
+       The name of the task that this step belongs to
+
+    yaml_filename : str
+       The name of the yaml file for this forward step
+
+    nu : float
+       The Laplacian viscosity to use for this forward step
+    """
+
+    def __init__(
+        self,
+        component,
+        init,
+        yaml_filename='forward.yaml',
+        name='forward',
+        task_name='default',
+        subdir=None,
+        indir=None,
+        ntasks=None,
+        min_tasks=None,
+        openmp_threads=1,
+        nu=1000.0,
+    ):
+        """
+        Create a new test case
+
+        Parameters
+        ----------
+        component : polaris.Component
+            The component the step belongs to
+
+        name : str
+            the name of the task
+
+        task_name : str
+           The name of the task that this step belongs to
+
+        yaml_filename : str
+           The name of the yaml file for this forward step
+
+        init : polaris.ocean.tasks.internal_wave.init.Init
+            the initial state step
+
+        subdir : str, optional
+            the subdirectory for the step.  The default is ``name``
+
+        ntasks : int, optional
+            the number of tasks the step would ideally use.  If fewer tasks
+            are available on the system, the step will run on all available
+            tasks as long as this is not below ``min_tasks``
+
+        min_tasks : int, optional
+            the number of tasks the step requires.  If the system has fewer
+            than this number of tasks, the step will fail
+
+        openmp_threads : int, optional
+            the number of OpenMP threads the step will use
+
+        nu : float, optional
+            the viscosity (if different from the default for the test group)
+        """
+        if min_tasks is None:
+            min_tasks = ntasks
+        super().__init__(
+            component=component,
+            name=name,
+            subdir=subdir,
+            indir=indir,
+            ntasks=ntasks,
+            min_tasks=min_tasks,
+            openmp_threads=openmp_threads,
+            update_eos=True,
+            graph_target=f'{init.path}/culled_graph.info',
+        )
+        self.task_name = task_name
+        self.yaml_filename = yaml_filename
+        self.nu = nu
+
+        # make sure output is double precision
+        self.add_yaml_file('polaris.ocean.config', 'output.yaml')
+
+        self.add_input_file(
+            filename='initial_state.nc',
+            work_dir_target=f'{init.path}/init.nc',
+        )
+
+        self.add_output_file(
+            filename='output.nc',
+            validate_vars=[
+                'layerThickness',
+                'normalVelocity',
+                'temperature',
+            ],
+        )
+
+    def dynamic_model_config(self, at_setup):
```

Comment:
Do we also need to use the same EOS parameters as specified in the cfg file at runtime?